From 2411f1fe5adc65c13296bf0320c2ea838a885226 Mon Sep 17 00:00:00 2001
From: Yujing Gu <guy@codeaurora.org>
Date: Tue, 10 Nov 2015 17:00:21 +0800
Subject: [PATCH] STK application should not show the STK menu for unsupported
 card.

When card doesn't support STK, STK application should not show the
STK menu.

CRs-Fixed: 937389

Change-Id: I6d329142202c6fb59672a74a894d2e8f3237ce24
---
 src/com/android/stk/StkLauncherActivity.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/stk/StkLauncherActivity.java b/src/com/android/stk/StkLauncherActivity.java
index a5271f6..ef4784f 100755
--- a/src/com/android/stk/StkLauncherActivity.java
+++ b/src/com/android/stk/StkLauncherActivity.java
@@ -176,6 +176,10 @@ public class StkLauncherActivity extends ListActivity {
         for (int i = 0; i < simCount; i++) {
             //Check if the card is inserted.
             if (mTm.hasIccCard(i)) {
+                if (appService == null || appService.getMainMenu(i) == null) {
+                    CatLog.d(LOG_TAG, "SIM " + i + " main menu of STK in the card is null");
+                    continue;
+                }
                 CatLog.d(LOG_TAG, "SIM " + i + " add to menu.");
                 mSingleSimId = i;
                 stkMenuTitle = appService.getMainMenu(i).title;
-- 
2.7.4

